### PR TITLE
mmtc: init at 0.2.12

### PR DESCRIPTION
**Pull request by figsoda**
###### Motivation for this change

mmtc is a minimal mpd terminal client that aims to be simple yet highly configurable

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -3065,6 +3065,12 @@
     githubId = 8182846;
     name = "Francesco Gazzetta";
   };
+  figsoda = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:3068]: Please put this into a separate commit with the message ``maintainers: add figsoda``.

File: pkgs/applications/audio/mmtc/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  src = fetchFromGitHub {
+    owner = "figsoda";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1raan98cljgw68zar490v01iqa8z8kk0if3j01b911j9rilir70r";
+  };
+  cargoSha256 = "1mx1d7bfxgyms91xah4hg28npsadqm5amkvmih3fp1gvmg6kqwrd";
+  meta = with lib; {
+    description =
+      "Minimal mpd terminal client that aims to be simple yet highly configurable";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mmtc/default.nix:15]:
```suggestion
  version = "0.2.9";

  src = fetchFromGitHub {
    owner = "figsoda";
    repo = pname;
    rev = "v${version}";
    sha256 = "1raan98cljgw68zar490v01iqa8z8kk0if3j01b911j9rilir70r";
  };

  cargoSha256 = "1mx1d7bfxgyms91xah4hg28npsadqm5amkvmih3fp1gvmg6kqwrd";

  meta = with lib; {
    description = "Minimal mpd terminal client that aims to be simple yet highly configurable";
```

File: pkgs/applications/audio/mmtc/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    homepage = "https://github.com/figsoda/mmtc";
+    changelog = "https://github.com/figsoda/mmtc/blob/v${version}/CHANGELOG.md";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ figsoda ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mmtc/default.nix:20]:
```suggestion
```

set by buildRustPackage.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/mmtc/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0hl98cfzlg2icfqgnpqasgrk6772wsraj78hd0k8slwa5i49n61i";
+  };
+
+  cargoSha256 = "01a8nd9fvh6w556yx88z52xp9b1l9r1qfw8n8ql9bn1l3mf1lkns";
+
+  meta = with lib; {
+    description =
+      "Minimal mpd terminal client that aims to be simple yet highly configurable";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mmtc/default.nix:18]:
```suggestion
    description = "Minimal mpd terminal client that aims to be simple yet highly configurable";
```

Comment (figsoda) [reply pkgs/applications/audio/mmtc/default.nix]: having a space at the end fails editorconfig check, is having them in the same line fine?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112070` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>mmtc</li>
</ul>
</details>

File: pkgs/applications/audio/mmtc/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
+{ callPackages, fetchFromGitHub, lib, rustPlatform, stdenv }:
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/mmtc/default.nix:1]:
```suggestion
{ fetchFromGitHub, lib, rustPlatform }:
```

Comment (figsoda) [reply pkgs/applications/audio/mmtc/default.nix]: Does it look good now?
